File: c/03-estrutura-repetitiva/divisao.c
/*Problema "divisao" (adaptado de URI 1116)
Escreva um algoritmo que leia dois números e imprima o resultado da divisão do primeiro pelo
segundo. Caso não for possível, mostre a mensagem “DIVISAO IMPOSSIVEL”.*/

#include <stdio.h> 
int main () {
    int n;
    double numerator, denominator, division;
    printf("Quantos casos voce vai digitar? ");
    scanf("%d", &n);

    for(int i = 1; i <= n; i++) {
        printf("Entre com o numerador: ");
        scanf("%lf", &numerator);
        printf("Entre com o denominador: ");
        scanf("%lf", &denominator);
        
        if (denominator == 0) {
            printf("DIVISAO IMPOSSIVEL\n");
        } else {
            division = numerator / denominator;
            printf("DIVISAO = %.2lf\n", division);
        }
    }
    return 0;
}

File: java/03-estruturas-repetitivas/Divisao.java
/*Problema "divisao" (adaptado de URI 1116)
Escreva um algoritmo que leia dois números e imprima o resultado da divisão do primeiro pelo
segundo. Caso não for possível, mostre a mensagem “DIVISAO IMPOSSIVEL”.*/

import java.util.Locale;
import java.util.Scanner;

public class Divisao {

	public static void main(String[] args) {
		Locale.setDefault(Locale.US);
		Scanner sc = new Scanner(System.in);
		
		int n;
		double a, b, c;
		
		System.out.print("Quantos casos voce vai digitar: ");
		n = sc.nextInt();
		
		for (int i = 1; i <= n; i++) {
			System.out.print("Entre com o numerador: ");
			a = sc.nextDouble();
			System.out.print("Entre com o denominador: ");
			b = sc.nextDouble();
					
			if (b == 0) {
				System.out.println("DIVISAO IMPOSSIVEL");
			} else {
				c = a / b;
				System.out.println("DIVISAO = " + String.format("%.2f", c));
			}
		}
		
		sc.close();
	}

}
